server/db/migrations/20171020144702_initial_schema.js

```diff
@@ -0,0 +1,41 @@
+exports.up = knex =>
+	knex.schema
+		.raw('create extension if not exists "uuid-ossp"')
+		.createTable('flair', table => {
+			table.increments('id').primary().notNullable()
+			table.string('name', 20).notNullable()
+			table.string('description', 100)
+			table.text('filename').notNullable()
+		})
+		.createTable('ball', table => {
+			table.increments('id').primary().notNullable()
+			table.string('name', 20).notNullable()
+			table.string('description', 100)
+			table.text('filename').notNullable()
+		})
+		.createTable('player', table => {
+			table.uuid('id').primary().defaultTo(knex.raw('uuid_generate_v4()')).notNullable()
+			table.dateTime('created_at').defaultTo(knex.fn.now()).notNullable()
+			table.dateTime('last_active').defaultTo(knex.fn.now()).notNullable()
+			table.string('username', 20).unique().notNullable()
+			table.dateTime('last_modified_username')
+			table.string('email', 254).unique()
+			table.string('password', 100).notNullable() // TODO update this length to reflect pw crypto
+			table.integer('selected_flair').references('id').inTable('flair').onDelete('SET NULL')
+			table.integer('selected_ball').references('id').inTable('ball').onDelete('SET NULL')
+			table.specificType('competitive_honors', 'text[]')
+		})
+		.createTable('map', table => {
```

Comment:
Perhaps maps should have author information. Also note, some maps have more than 1 author

Reply:
Good shout, I'll add an authors key as something like `varchar(n)[]`.

Worth adding the same to flairs and balls or no?

Reply:
Since when do flairs and balls have authors? Or are you implying something else?

Reply:
I'm imagining a system whereby flairs/balls/other assets are contributed to the game by the community. In that case they may like accreditation as map authors do.

Reply:
How do we want authors to be denoted? I was thinking an array of strings, but then I wondered if we might prefer actually linking them to the players i.e. an array of player IDs (corresponding to the player table) instead.